pkgs/applications/audio/musescore/default.nix

```
@@ -83,6 +83,8 @@ in stdenv'.mkDerivation (finalAttrs: {
     # Don't bundle qt qml files, relevant really only for darwin, but we set
     # this for all platforms anyway.
     "-DMUE_COMPILE_INSTALL_QTQML_FILES=OFF"
+    # Don't build unit tests unless we are going to run them.
+    (lib.cmakeBool "MUSE_ENABLE_UNIT_TESTS" finalAttrs.doCheck)
```

```suggestion
    (lib.cmakeBool "MUSE_ENABLE_UNIT_TESTS" finalAttrs.finalPackage.doCheck)
```

(Yes there are anti-patterns in other places)

(And yes the documentation isn't clear about this)

https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes

---

Wow, thanks!

---

TLDR: the default value of `doCheck` is determined outside of that fixed point, between the fixed point and actually attributes passed to `builtins.derivation`. I'll try to solve this